From 43d15003392a1ab45a03dc651a9c0dacbdfe4a8a Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz@ettus.com>
Date: Wed, 25 Apr 2018 20:06:55 -0700
Subject: [PATCH] misc: eeprom: Set io_limit to 64 to avoid overflowing
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Set the io_limit to 16 to avoid overflowing the i2c
passthrough through the MCU. This should properly be
fixed propably with a device_propery pulled out of
the devicetree.

Signed-off-by: Moritz Fischer <moritz@ettus.com>

Increase io_limit to 64 because this was the default
that was used in the U-Boot bootargs variable for
x4xx for a long time and which proofed to be
reliable.

Signed-off-by: Jörg Hofrichter <joerg.hofrichter@emerson.com>
---
 drivers/misc/eeprom/at24.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/eeprom/at24.c b/drivers/misc/eeprom/at24.c
index e88d61b844242..109544762d785 100644
--- a/drivers/misc/eeprom/at24.c
+++ b/drivers/misc/eeprom/at24.c
@@ -105,7 +105,7 @@ struct at24_data {
  *
  * This value is forced to be a power of two so that writes align on pages.
  */
-static unsigned int at24_io_limit = 128;
+static unsigned int at24_io_limit = 64;
 module_param_named(io_limit, at24_io_limit, uint, 0);
 MODULE_PARM_DESC(at24_io_limit, "Maximum bytes per I/O (default 128)");
 
-- 
2.43.0

